lib/src/realm_object.dart

```diff
@@ -203,7 +204,9 @@ extension RealmEntityInternal on RealmEntity {
 mixin RealmObject on RealmEntity {
   RealmObjectHandle? _handle;
   RealmAccessor _accessor = RealmValuesAccessor();
-  static final Map<Type, RealmObject Function()> _factories = <Type, RealmObject Function()>{};
+  static final Map<Type, RealmObject Function()> _factories = <Type, RealmObject Function()>{
+    RealmObject: () => DynamicRealmObject._(),
```

Comment:
Not sure why would we need this.

Reply:
Because we need to make sure that `List<RealmObject>` will contain `DynamicRealmObject` elements.